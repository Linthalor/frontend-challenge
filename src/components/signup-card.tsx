import { Box, Card, CardActions, CardContent, useMediaQuery } from '@mui/material';
import { theme } from '../app/theme';

export const SignUpCard = ({
  content,
  actions
}: {
  content: JSX.Element,
  actions: JSX.Element
}) => {
  const fullscreenSignUpCardMediaQuery = useMediaQuery(theme.breakpoints.down('sm'));
  return (
    <Box
      sx={{
        width: '100%',
        height: '100%',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
      }}
    >
      <Card
        sx={{
          width: fullscreenSignUpCardMediaQuery ? '100%' : 400,
          height: fullscreenSignUpCardMediaQuery ? '100%' : 'unset',
          display: 'flex',
          flexDirection: 'column',
          justifyContent: 'center',
        }}
      >
        <CardContent
          sx={{
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'stretch',
            justifyContent: 'start',
            gap: 1,
            flex: fullscreenSignUpCardMediaQuery ? 0 : 1,
          }}
        >
          {content}
        </CardContent>
        <CardActions
          sx={{
            display: 'flex',
            flexDirection: 'row',
            justifyContent: 'end',
            gap: 1,
          }}
        >
          {actions}
        </CardActions>
      </Card>
    </Box>
  );
};