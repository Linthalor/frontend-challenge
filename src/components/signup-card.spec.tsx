import { cleanup, render } from '@testing-library/react';
import { SignUpCard } from './signup-card';

jest.mock('@mui/material');

describe('SignUpCard', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders as expected', () => {
    const MockContent = 'mocked-content';
    const MockActions = 'mocked-actions';

    const { container } = render(
      <SignUpCard content={<MockContent />} actions={<MockActions />} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});