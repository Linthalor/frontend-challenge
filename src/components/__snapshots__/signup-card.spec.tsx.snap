// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpCard renders as expected 1`] = `
<mocked-mui-box
  sx="{\\"width\\":\\"100%\\",\\"height\\":\\"100%\\",\\"display\\":\\"flex\\",\\"justifyContent\\":\\"center\\",\\"alignItems\\":\\"center\\"}"
>
  <mocked-mui-card
    sx="{\\"width\\":400,\\"height\\":\\"unset\\",\\"display\\":\\"flex\\",\\"flexDirection\\":\\"column\\",\\"justifyContent\\":\\"center\\"}"
  >
    <mocked-mui-card-content
      sx="{\\"display\\":\\"flex\\",\\"flexDirection\\":\\"column\\",\\"alignItems\\":\\"stretch\\",\\"justifyContent\\":\\"start\\",\\"gap\\":1,\\"flex\\":1}"
    >
      <mocked-content />
    </mocked-mui-card-content>
    <mocked-mui-card-actions
      sx="{\\"display\\":\\"flex\\",\\"flexDirection\\":\\"row\\",\\"justifyContent\\":\\"end\\",\\"gap\\":1}"
    >
      <mocked-actions />
    </mocked-mui-card-actions>
  </mocked-mui-card>
</mocked-mui-box>
`;
